services/wallet/activity/service.go

```diff
@@ -22,21 +22,22 @@ const (
 	EventActivityGetOldestTimestampDone walletevent.EventType = "wallet-activity-get-oldest-timestamp-result"
 )
 
-var (
-	filterTask = TaskType{
-		ID:     1,
-		Policy: ReplacementPolicyCancelOld,
-	}
-	getRecipientsTask = TaskType{
-		ID:     2,
-		Policy: ReplacementPolicyIgnoreNew,
-	}
-	getOldestTimestampTask = TaskType{
-		ID:     3,
-		Policy: ReplacementPolicyCancelOld,
-	}
+const (
+	filterTaskID                 = int32(1)
+	filterTaskPolicy             = ReplacementPolicyCancelOld
+	getRecipientsTaskID          = int32(2)
+	getRecipientsTaskPolicy      = ReplacementPolicyIgnoreNew
+	getOldestTimestampTaskID     = int32(3)
+	getOldestTimestampTaskPolicy = ReplacementPolicyCancelOld
 )
 
+func makeTaskType(requestID int32, originalID int32, policy ReplacementPolicy) TaskType {
+	return TaskType{
+		ID:     int64(requestID)<<32 | int64(originalID),
```

Comment:
> Is there any benefit on using single field here instead of using 2?

Mainly the simplicity of implementation. There is a need for a unique identity for each task that has to be queued. Brings no benefit to having two fields for the scheduler. I see no benefit in passing the request-id responsibility to the generic queued scheduler.

> My opinion is that it makes it less obvious for a reader that it has 2 meanings.

Unicity is a combination of request ID and task ID which is defined by the service requirements, and it is the responsibility of the caller, `activity.Service` in this case.
Extending the scheduler with these two IDs won't bring any benefit outside the bigger space for each identifier that is unnecessary for this implementation. I mean both ID types have local meanings: request ID represents the unique clients (two for now) and task ID represents the unique tasks supported by the service (three for now). 

> And reverse operation is implicit right?

Because of the only unique identities requirements, there is no need to extract back the values, they just need to provide uniqueness; see `service_test.go/Test_makeTaskType`

Does it make sense? How do you think I can clarify this without complicating the implementation?